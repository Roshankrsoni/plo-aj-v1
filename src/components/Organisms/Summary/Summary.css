.mainlogo {
  opacity: 0;
  width: 20rem;
  height: 20rem;
  border-radius: 50%;
  transition: opacity 0.3s ease-in-out;
  /* animation: circularSpin 5s infinite ease-in; */
}

.mainlogo-wrapper {
  position: relative;
  box-sizing: content-box;
}

.mainlogo-wrapper:after {
  content: "";
  position: absolute;
  width: 20rem;
  height: 20rem;
  border: 2px solid var(--red);
  border-radius: 50%;
  top: 50%;
  left: 50%;
  transform: translateX(-50%) translateY(-50%);
  animation: borders 6s infinite linear;
}

.mainlogo-wrapper:before {
  content: "";
  position: absolute;
  width: 20rem;
  height: 20rem;
  border: 2px solid var(--red);
  border-radius: 50%;
  top: 50%;
  left: 50%;
  transform: translateX(-50%) translateY(-50%);
  animation: borders 6s infinite 3s linear;
  opacity: 0;
}

.loaded {
  opacity: 1;
}

@keyframes borders {
  0% {
    width: 20rem;
    height: 20rem;
    opacity: 1;
  }
  100% {
    width: 24rem;
    height: 24rem;
    opacity: 0;
  }
}
