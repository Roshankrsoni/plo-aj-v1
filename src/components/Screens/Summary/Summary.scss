@keyframes borders {
  0% {
    width: calc(var(--image-width) + 1rem);
    height: calc(var(--image-width) + 1rem);
    border: 2px solid var(--red);
  }
  100% {
    width: calc(var(--image-width) + 4rem);
    height: calc(var(--image-width) + 4rem);
    border: 2px solid transparent;
  }
}

.summary-mobile {
  align-items: center;
  display: flex;
  flex-direction: column;
  --image-width: 16rem;

  .mainlogo-wrapper {
    position: relative;
    box-sizing: content-box;

    &:after {
      content: "";
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translateX(-50%) translateY(-50%);
      width: var(--image-width);
      height: var(--image-width);
      border-radius: 50%;
      animation: borders 6s infinite linear;
    }

    &:before {
      content: "";
      position: absolute;
      width: var(--image-width);
      height: var(--image-width);
      border-radius: 50%;
      top: 50%;
      left: 50%;

      transform: translateX(-50%) translateY(-50%);
      animation: borders 6s infinite 3s linear;
    }

    .mainlogo {
      opacity: 0;
      width: var(--image-width);
      height: var(--image-width);
      border-radius: 50%;
      transition: opacity 0.3s ease-in-out;

      &.loaded {
        opacity: 1;
      }
    }
  }

  .main-title {
    padding-top: 0;
  }
  .summary-text {
    padding-top: 1.5rem;
    span {
      font-weight: bold;
      color: var(--red);
    }

    span:nth-child(1) {
      background: var(--red);
      padding: 0 0.5rem;
      color: var(--grey0);
      font-weight: bold;
    }
  }

  .hello {
    padding-top: 3.75rem;
    letter-spacing: 1px;
  }
}
