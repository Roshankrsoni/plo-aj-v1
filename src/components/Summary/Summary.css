.mainlogo{
    transition: all 1s ease-in-out ;
    animation: zoomIn 1s ease-in-out forwards;
}

.mainlogo:hover .eyebrow.right{

    transform-origin: 50% 50%;
    animation: eyebrowmovementRight 0.25s ease-in-out 2;
} 
 .mainlogo:hover .eyebrow.left{
    transform-origin: 50% 50%;
    animation: eyebrowmovementLeft 0.25s ease-in-out 2;
}

.darkmode .mainlogo{
    filter: invert(94%) sepia(93%) saturate(0%) hue-rotate(122deg) brightness(103%) contrast(90%);
}

@keyframes eyebrowmovementRight{
    0%{
        transform: translateY(0);
    }
    80%{
        transform: rotate(6deg) translateY(-5px);

    }
    100%{
        transform: translateY(0);
    }
}
@keyframes eyebrowmovementLeft{
    0%{
        transform: translateY(0);
    }
    80%{
        transform: rotate(-6deg) translateY(-5px);

    }
    100%{
        transform: translateY(0);
    }
}


@keyframes zoomIn{
    0%{
        transform: scale(0);
    }
    80%{
        transform: scale(1.1);
    }
    100%{
        transform: scale(1);
    }
}



