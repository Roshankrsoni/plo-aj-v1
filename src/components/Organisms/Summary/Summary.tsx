import React, { useMemo, useState } from "react";
import lightImage from "../../../assets/portfolio-app-img-light-min.png";
import darkImage from "../../../assets/portfolio-app-img-dark-min.png";
import "./Summary.css";
import Loader from "../../Atoms/Loader/Loader";
import { useGlobalContext } from "../../../context/GlobalContext";
import { useTheme } from "../../../context/ThemeContext";
import moment from "moment/moment";

const Summary = () => {
  const [imageLoading, setImageLoading] = useState(false);
  const { profileData } = useGlobalContext();
  const { darkMode } = useTheme();
  const { overview, experience } = profileData;

  const totalExperience = useMemo(() => {
    return experience
      .map((exp) => {
        const fromDate = moment(exp.fromDate);
        const toDate = exp.toDate !== "Present" ? moment(exp.toDate) : moment();
        return moment(toDate).diff(fromDate, "year", true);
      })
      .reduce((a, b) => a + b)
      .toFixed(1);
  }, [experience]);

  function handleImageLoaded() {
    setImageLoading(true);
  }
  function handleImageErrored() {
    console.error("Error while loading the profile image");
    setImageLoading(false);
  }
  return (
    <>
      <div className="col-md-7 page-1 text-center ">
        <div className="mainlogo-wrapper">
          <img
            src={darkMode ? darkImage : lightImage}
            className={`mainlogo ${imageLoading && "loaded"}`}
            alt="mi Baburao"
            loading="eager"
            onLoad={handleImageLoaded}
            onError={handleImageErrored}
          />
        </div>

        <div className="h1 bold main-title grey-1">{overview.name}</div>
        <div className="h4 uppercase letterspacing-1 red bold">
          {overview.title}
        </div>
        <div className="body-text summary-text">
          {overview.summary.replace(
            "{totalYearsOfExperience}",
            totalExperience
          )}
        </div>
        <div className="default-text red scroll-text thin">
          Scroll to Know more
        </div>
      </div>
    </>
  );
};

export default Summary;
