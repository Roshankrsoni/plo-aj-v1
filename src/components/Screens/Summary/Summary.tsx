import React, { useMemo, useState } from "react";
import lightImage from "../../../assets/portfolio-app-img-light-min.png";
import darkImage from "../../../assets/portfolio-app-img-dark-min.png";
import "./Summary.scss";
import { useGlobalContext } from "../../../context/GlobalContext";
import { useTheme } from "../../../context/ThemeContext";
import moment from "moment/moment";
import SectionVisibilityHOC from "../../Organisms/SectionVisibilityHOC/SectionVisibilityHOC";
import T from "../../../translations/en_IN";
import constants from "../../../helpers/constants";
import DownloadCV from "../../Molecules/DownloadCV/DownloadCV";
import parse from "html-react-parser";
const Summary = () => {
  const [imageLoading, setImageLoading] = useState(false);
  const { profileData } = useGlobalContext();
  const { darkMode } = useTheme();
  const { overview, experience } = profileData;

  const totalExperience = useMemo((): string => {
    try {
      return experience
        .map((exp) => {
          const fromDate = moment(exp.fromDate);
          const toDate =
            exp.toDate !== "Present" ? moment(exp.toDate) : moment();
          return moment(toDate).diff(fromDate, "year", true);
        })
        .reduce((a, b) => a + b)
        .toFixed(1);
    } catch (err) {
      return String(moment().year() - 2014);
    }
  }, [experience]);

  function handleImageLoaded() {
    setImageLoading(true);
  }
  function handleImageErrored() {
    console.error("Error while loading the profile image");
    setImageLoading(false);
  }

  return (
    <div className="summary-mobile">
      <div className="mainlogo-wrapper">
        <img
          src={darkMode ? darkImage : lightImage}
          className={`mainlogo ${imageLoading && "loaded"}`}
          alt="mi Baburao"
          loading="eager"
          onLoad={handleImageLoaded}
          onError={handleImageErrored}
        />
      </div>

      <div className="bold primary-color hello">{T.HELLO}</div>
      <div className="h1 bold main-title grey-1 typewriter">
        {overview.name}
      </div>
      <div className="body-text summary-text">
        {parse(
          overview.summary.replace("{totalYearsOfExperience}", totalExperience)
        )}
      </div>
      <DownloadCV />
    </div>
  );
};

export default SectionVisibilityHOC(Summary, constants.classNames.SUMMARY);
