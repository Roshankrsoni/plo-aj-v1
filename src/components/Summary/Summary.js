import React, { useState, useEffect } from "react";
import { useDispatch } from "react-redux";
import myImage from "../../assets/AnoopJadhav.png";
import "./Summary.css";
import Loader from '../Loader/Loader'

export default ({ dbData }) => {

  const [imageLoading, setImageLoading] = useState(false);

  function handleImageLoaded() {
    setImageLoading(true)
  }

  function handleImageErrored() {
    setImageLoading(false)
  }

  return (
    <>
      {!imageLoading && <Loader />}
      <div
        className={
          "show-on-scroll col-md-7 page-1 text-center "
        }
      >
        <div className="mainlogo-wrapper">
          <img
            src={myImage}
            className={`mainlogo ${imageLoading && 'loaded'}`}
            alt="mi Baburao"
            loading="eager"
            onLoad={handleImageLoaded}
            onError={handleImageErrored}
          />
        </div>

        <div className="h1 bold main-title grey-1">
          {dbData.overview.name}
        </div>
        <div className="h4 uppercase letterspacing-1 red bold">
          {dbData.overview.title}
        </div>
        <div className="body-text summary-text">
          {dbData.overview.summary}
        </div>
        <div className="default-text red scroll-text thin">
          Scroll to Know more
        </div>
      </div>
    </>
  );
}
