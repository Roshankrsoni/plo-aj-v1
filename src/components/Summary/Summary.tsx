import React, { useState } from "react";
import myImage from "../../assets/AnoopJadhav.png";
import "./Summary.css";
import Loader from "../Loader/Loader";
import { useGlobalContext } from "../../context/GlobalContext";

const Summary = () => {
  const [imageLoading, setImageLoading] = useState(false);
  const { profileData } = useGlobalContext();
  const { overview } = profileData;
  function handleImageLoaded() {
    setImageLoading(true);
  }
  function handleImageErrored() {
    console.error("Error while loading the profile image");
    setImageLoading(false);
  }
  return (
    <>
      {!imageLoading && <Loader />}
      <div className="col-md-7 page-1 text-center ">
        <div className="mainlogo-wrapper">
          <img
            src={myImage}
            className={`mainlogo ${imageLoading && "loaded"}`}
            alt="mi Baburao"
            loading="eager"
            onLoad={handleImageLoaded}
            onError={handleImageErrored}
          />
        </div>

        <div className="h1 bold main-title grey-1">{overview.name}</div>
        <div className="h4 uppercase letterspacing-1 red bold">
          {overview.title}
        </div>
        <div className="body-text summary-text">{overview.summary}</div>
        <div className="default-text red scroll-text thin">
          Scroll to Know more
        </div>
      </div>
    </>
  );
};

export default Summary;
