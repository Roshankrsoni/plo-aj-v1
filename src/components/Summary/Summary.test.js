
import React from 'react';
import { shallow } from 'enzyme';
import Summary from './Summary';

import dummyData from '../TestData/testData.json'

describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Summary dbData={dummyData} />);
        expect(component).toMatchSnapshot();
    });
});