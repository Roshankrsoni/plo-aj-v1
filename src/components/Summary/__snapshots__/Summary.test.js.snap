// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="show-on-scroll col-md-7 page-1 text-center undefined"
>
  <img
    alt="mi Baburao"
    className="mainlogo loading"
    loading="eager"
    onError={[Function]}
    onLoad={[Function]}
    src="AnoopJadhav.png"
  />
  <div
    className="h1 bold main-title grey-1"
  >
    Anoop Jadhav
  </div>
  <div
    className="h4 uppercase letterspacing-1 red bold"
  >
    UI DEVELOPER
  </div>
  <div
    className="body-text summary-text"
  >
    A passionate UI Developer with around 6 years of experience using tools and technologies like HTML, CSS, JavaScript, React Js, Vue Js, Salesforce Lightning Components, Salesforce LWC,  Adobe Illustrator, Adobe XD, Figma, and Sketch.
  </div>
  <div
    className="default-text red scroll-text thin"
  >
    Scroll to Know more
  </div>
</div>
`;
